File: cdmsbats/TLBRecord.py
# -*- coding: utf-8 -*-
"""
Created on Fri Apr 11 12:14:28 2014

@author: tdoughty1
"""

from numpy import copy
from DataRecord import DataRecord


class TLBRecord(DataRecord):

    def __init__(self):

        self._TriggerTime = None
        self._TriggerMask = None

    def StoreValues(self, Record):

        self._TriggerTime = Record[0]
        self._TriggerMask = copy(Record[1:7])

    def PrintValues(self):

        print "Trigger Time = %d" % self._TriggerTime
        for i, mask in zip(range(len(self._TriggerMask)), self._TriggerMask):
            print "Tower %d TLB Mask = 0x%x" % (i+1, mask)

File: cdmsbats/tlbrecord.cpp
#include <iostream>
#include <iomanip>

#include "tlbrecord.h"

using namespace std;

TLBRecord::TLBRecord()
{
    _TriggerMask.resize(6);
}

void TLBRecord::StoreValues(uint32_t* Record)
{
    for(int i=0; i<6; i++)
    {
        _TriggerMask[i] = Record[i];
    }
}

void TLBRecord::PrintValues()
{
    for(int i=0; i<6; i++)
    {
        cout << "Tower " << (i+1) << " TLB Mask = 0x" << setbase(16) << _TriggerMask[i] << endl;
    }
}
